spec/support/controller_macros.rb

```diff
@@ -0,0 +1,16 @@
+module ControllerMacros
+  def login_admin
+    before(:each) do
+      @request.env["devise.mapping"] = Devise.mappings[:admin]
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.